exasol/bucketfs/_path.py

```diff
@@ -0,0 +1,378 @@
+from __future__ import annotations
+from typing import Protocol, ByteString, BinaryIO, Iterable, Generator, Optional
+from pathlib import PurePath, PureWindowsPath
+import errno
+import os
+from io import IOBase
+from exasol.bucketfs._buckets import BucketLike
+
+
+class PathLike(Protocol):
```

Comment:
I thought we already added the PathLike protocol in https://github.com/exasol/bucketfs-python/pull/110/files

Reply:
I just moved it to the same file.